src/server/scripts/EasternKingdoms/Deadmines/boss_vancleef.cpp

```diff
@@ -0,0 +1,142 @@
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "Deadmines.h"
+
+enum VanCleefData
+{
+    SPELL_DUAL_WIELD = 674,
+    SPELL_THRASH = 12787,
+    SPELL_VANCLEEFS_ALLIES = 5200,
+
+    PHASE_ONE    = 1,
+    PHASE_66_PCT = 2,
+    PHASE_50_PCT = 3,
+    PHASE_33_PCT = 4,
+    PHASE_25_PCT = 5
+};
+
+enum Speech
+{
+    SAY_AGGRO  = 0,    // None may challenge the Brotherhood!
+    SAY_ONE    = 1,    // Lapdogs, all of you!
+    SAY_SUMMON = 2,    // calls more of his allies out of the shadows.
+    SAY_TWO    = 3,    // Fools! Our cause is righteous!
+    SAY_KILL   = 4,    // And stay down!
+    SAY_THREE  = 5     // The Brotherhood shall prevail!
+};
+
+// From TDB 335.63
+Position const BlackguardPositions[] =
+{
+        { -78.2791f, -824.784f, 40.0007f, 2.93215f },
+        { -77.8071f, -815.097f, 40.0188f, 3.26377f }
+};
+
+struct boss_vancleef : public BossAI
+{
+    public:
+        boss_vancleef(Creature* creature) : BossAI(creature, DATA_VANCLEEF), _phase(PHASE_ONE){ }
+
+        void Reset() override
+        {
+            BossAI::Reset();
+
+            _phase = PHASE_ONE;
+
+            DoCastSelf(SPELL_DUAL_WIELD, true);
+            DoCastSelf(SPELL_THRASH, true);
+
+            SummonBlackguards();
+        }
+
+        void EnterCombat(Unit* victim) override
+        {
+            BossAI::EnterCombat(victim);
+            summons.DoZoneInCombat();
+
+            Talk(SAY_AGGRO);
+        }
+
+        void KilledUnit(Unit* victim) override
+        {
+            if (victim->GetTypeId() == TYPEID_PLAYER)
+                Talk(SAY_KILL);
+        }
+
+        void EnterEvadeMode(EvadeReason /*why*/) override
+        {
+            summons.DespawnAll();
+            _DespawnAtEvade();
+        }
+
+        void SummonBlackguards()
+        {
+            for (Position BlackguardPosition : BlackguardPositions)
+                DoSummon(NPC_BLACKGUARD, BlackguardPosition, 60000, TEMPSUMMON_CORPSE_TIMED_DESPAWN);
+        }
+
+        void SwitchPhase(uint8 Phase)
+        {
+            // don't repeat
+            if (_phase == Phase)
+                return;
+
+            _phase = Phase;
+
+            switch (_phase)
+            {
+                case PHASE_66_PCT:
+                    Talk(SAY_ONE);
+                    break;
+                case PHASE_50_PCT:
+                    Talk(SAY_SUMMON);
+                    DoCastSelf(SPELL_VANCLEEFS_ALLIES);
+                    break;
+                case PHASE_33_PCT:
+                    Talk(SAY_TWO);
+                    break;
+                case PHASE_25_PCT:
+                    Talk(SAY_THREE);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        void DamageTaken(Unit* /*attacker*/, uint32& /*damage*/) override
+        {
+            if (_phase == PHASE_ONE && HealthBelowPct(66))
```

Comment:
HealthBelowPctDamaged is better
Also I'd change the order of the checks, example scenario: a level 80 uses a spell that brings the boss to 5% health, which phase should be triggered? One or Last?

Reply:
And also, DamageTaken is NOT a standard in TC for phase switching
I personally prefer to check it on update since DamageTaken can be called multiple times on the same update tick, from different damage incomes, giving the chance to multiple same triggered actions

Reply:
@ccrs I wondered the same about hp % at https://github.com/TrinityCore/TrinityCore/pull/20736#issuecomment-340165344

Reply:
you can control multiple calls using a variable, i dont see any problem about it.
But i agree with you, is better change the order.

Reply:
A little late but IIRC blizz uses update to check hp